next/lib/drupal/drupal-client.ts

```diff
@@ -23,12 +23,31 @@ const getFetcher =
     });
   };
 
+// This instance of the client will connect to Drupal using a consumer that
+// is associated with a role with "regular" permissions. It should be used by
+// default.
 export const drupal = new GraphQlDrupalClient(env.NEXT_PUBLIC_DRUPAL_BASE_URL, {
   // We specify our own fetcher to apply pRetry to it:
   fetcher: getFetcher(),
   forceIframeSameSiteCookie: env.NODE_ENV === "development",
   auth: {
-    clientId: env.DRUPAL_CLIENT_ID,
-    clientSecret: env.DRUPAL_CLIENT_SECRET,
+    clientId: env.DRUPAL_CLIENT_VIEWER_ID,
+    clientSecret: env.DRUPAL_CLIENT_VIEWER_SECRET,
   },
 });
+
+// This instance of the client will connect to the Drupal API using a consumer
+// which is associated with a role with additional permissions. Use this instance
+// when you need to get data for unpublished nodes, like in previews.
+export const drupalPreviewer = new GraphQlDrupalClient(
+  env.NEXT_PUBLIC_DRUPAL_BASE_URL,
+  {
+    // We specify our own fetcher to apply pRetry to it:
+    fetcher: getFetcher(),
+    forceIframeSameSiteCookie: env.NODE_ENV === "development",
+    auth: {
+      clientId: env.DRUPAL_CLIENT_ID,
+      clientSecret: env.DRUPAL_CLIENT_SECRET,
+    },
+  },
+);
```

Comment:
I think we have to do it this way (creating two instances) instead of passing a different auth each time like the docs would allow, because our doGrapqhlQuery method does not have access to that information since it's marked as private in the class...) 

This way seems to work :-)